src/main/java/org/kairosdb/core/queue/MemoryQueueProcessor.java

```diff
@@ -40,16 +41,17 @@ public MemoryQueueProcessor(
 
     @Override
     public void addReportedMetrics(final PeriodicMetrics periodicMetrics) {
+        periodicMetrics.recordGauge("queue/persisted_samples", m_persistedSamplesCount.getAndSet(0L));
+        periodicMetrics.recordGauge("queue/dropped_samples", m_droppedSamplesCount.getAndSet(0L));
         periodicMetrics.recordGauge("queue/process_count", m_readFromQueueCount.getAndSet(0));
-        periodicMetrics.recordGauge("queue/memory_queue.size", m_queue.size());
+        periodicMetrics.recordGauge("queue/memory_queue_size", m_queue.size());
     }
 
     @Override
     public void put(final DataPointEvent dataPointEvent) {
-        try {
-            m_queue.put(dataPointEvent);
-        } catch (final InterruptedException e) {
-            logger.error("Error putting data", e);
+        if (!m_queue.offer(dataPointEvent)) {
```

Comment:
Just a note that this looks to have slightly different behavior than previously because put is blocking but offer is not. Just checking that that is desired behavior

Reply:
Yes, this is intentional. We are seeing lock ups up the stack and it's possible this is starving the write queue by blocking. To get an accurate measure of the number of dropped samples due to queue full we need to stop blocking.